### PR TITLE
Catalog handling to filter plans/services in case of cross consumption

### PR DESCRIPTION
**Pull request by subhankarc**
* Parameterized the path that gets registered as broker.
* Added settings for supported_platform and osbapi_compliant_name
* Catalogs are shown based on the supported_platform
* Changed implementation for PlatformManagers, instance assignment is removed from it.
* Added new test for K8S catalog
* Fixed Quota test which was changing the catalog
* Removed the unnecessary CC API version check

Also, see https://github.com/openservicebrokerapi/servicebroker/issues/442 and https://github.com/openservicebrokerapi/servicebroker/pull/452

### CODE REVIEW

File: lib/fabrik/Fabrik.js [hunk 1]
```diff
@@ -41,11 +42,20 @@ class Fabrik {
...
         return Promise
           .try(() => context ? context : instance.platformContext)
-          .then(context => instance.assignPlatformManager(BasePlatformManager.getInstance(instance_id, context)))
+          .then(context => instance.assignPlatformManager(Fabrik.getPlatformManager(context.platform)))
           .return(instance);
       });
   }
 
+  static getPlatformManager(platform) {
+    const PlatformManager = (platform && CONST.PLATFORM_MANAGER[platform]) ? require(`./${CONST.PLATFORM_MANAGER[platform]}`) : undefined;
```
Comment (kamath-prasad) [inline lib/fabrik/Fabrik.js]: attribute name should start with small case here - `platformManager`

File: config/settings.yml [hunk 2]
```diff
@@ -432,6 +432,9 @@ defaults: &defaults
       secret: 'service-fabrik-blueprint-secret'
```
Comment (kamath-prasad) [inline config/settings.yml]: maybe its better to have a new plan to have support for k8s and leave the current one's for cf as the plan name in k8s has to be in different format ?!

Comment (subhankarc) [reply config/settings.yml]: There will not be any new plan. The same existing BOSH plans will be shown for K8S as well.

File: lib/fabrik/Fabrik.js (hunk 1 above)
Comment (subhankarc) [reply lib/fabrik/Fabrik.js]: As it is a class reference, kept it as `PlatformManager`

File: lib/controllers/ServiceBrokerApiController.js [hunk 3]
```diff
@@ -34,20 +33,13 @@ class ServiceBrokerApiController extends BaseController {
         } else {
           throw new PreconditionFailed(`At least Broker API version ${minVersion} is required.`);
         }
```
[the remark this answers is not in the document]
Comment (jagadish-kb) [reply lib/controllers/ServiceBrokerApiController.js]: Post discussion, this is not needed. Keeping this comment for history.

File: lib/fabrik/BasePlatformManager.js [hunk 4]
```diff
@@ -1,15 +1,12 @@
...
-  constructor(guid, context) {
-    this.guid = guid;
-    this.context = context;
+  constructor(platform) {
+    this.platform = platform;
   }
 
-  get platform() {
-    return this.context.platform;
+  getPlatformSpecificCatalog(catalog) {
```
Comment (jagadish-kb) [inline lib/fabrik/BasePlatformManager.js]: Platformmanager is to return things platformSpecifc :-), so may be can we strip this from the method name and just keep it `getCatalog`. Also can return `undefined` from BasePlatformManager as its implementation from specific manager is absent.

File: lib/constants.js [hunk 5]
```diff
@@ -180,8 +180,16 @@ module.exports = Object.freeze({
...
   },
 
+  PLATFORM_ALIAS: {
+    CF: 'cf',
+    K8S: 'k8s'
+  },
+
   PLATFORM_MANAGER: {
-    'cloudfoundry': 'CfPlatformManager'
+    'cloudfoundry': 'CfPlatformManager',
```
[the remark this answers is not in the document]
Comment (subhankarc) [reply lib/constants.js]: I have change this and removed the duped ones. Have handled that in the `getPlatformManager` code.

File: lib/controllers/ServiceBrokerApiController.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (subhankarc) [reply lib/controllers/ServiceBrokerApiController.js]: sure.

File: lib/fabrik/BasePlatformManager.js (hunk 4 above)
Comment (subhankarc) [reply lib/fabrik/BasePlatformManager.js]: :) changed it.

**Pull request merged**